### PR TITLE
washing machine addition

### PR DESCRIPTION
**Pull request by slegars56**
- new sensors
- handling error codes (more error encryption codes to be added)

### CODE REVIEW

File: hon.py [hunk 1]
```diff
@@ -256,6 +256,8 @@ async def async_get_state(self, mac, typeName, returnAllData = False, loop=False
                 return json.loads(text)
             
             json_data = json.loads(text)["payload"]["shadow"]["parameters"]
```
Comment (gvigroux) [inline hon.py:258]: I don't know the impact of this on other integrations. Can you do it in another way?

File: hon.py [hunk 2]
```diff
@@ -256,6 +256,8 @@ async def async_get_state(self, mac, typeName, returnAllData = False, loop=False
...
+            json_data_lastCon = json.loads(text)["payload"]["lastConnEvent"]
```
Comment (gvigroux) [inline hon.py:259]: I don't know the impact of this on other integrations. Can you do it in another way?

**Comment by buhralex** [comment]
Thanks @slegars56! Your fork is the only one that's working with my new Haier Series 5. Also great addition of sensors!

File: hon.py (hunk 2 above)
Comment (dennisnn87) [reply hon.py:259]: Any way to be able to start/stop the machine from HA?

**Comment by dennisnn87** [comment]
All my sensors show up as unknown :/ hoover wash 500. 
Any way to start/stop/pause a job?

**Comment by dimenco-chris** [comment]
> Thanks @slegars56! Your fork is the only one that's working with my new Haier Series 5. Also great addition of sensors!

Same here. Thanks!

I'll definitely be making some changes of my own to try and get the remote start to work when I have time!

**Comment by Jaykurb** [comment]
> Thanks @slegars56! Your fork is the only one that's working with my new Haier Series 5. Also great addition of sensors!

Same here. Thank you @slegars56!

The following sensors show as unknown though:

- Spin speed
- Temperature
- Time Remaining

The current Electricity Used stayed at 0 so I assume this isn't working correctly. 

But nothing was available before making these changes. I can now finally set up automation for when the washing cycle has finished. It would be nice to get more sensors going though, any advice on how to get these running too?

**Comment by gvigroux** [comment]
Hi @slegars56, can you propose a new PR ? I cannot merge this one directly. Too many changes since you propose the it (sorry for that)

**Comment by slegars56** [comment]
No worries, I try to take a look this evening

**Comment by gvigroux** [comment]
> @gvigroux , No worries, I try to take a look this evening

thanks :)

**Comment by slegars56** [comment]
@gvigroux, I've updated my repository, maybe you want to test it before I create a PR? I tested only with my washing machine.

**Pull request closed without merging**